Implement transactions list

selankon opened the pull request.

selankon committed:
Implement transactions list

selankon committed:
Implement transactions detail

selankon committed:
Add some translations

elboletaire commented on src/queries/stats.ts:
```diff
@@ -14,3 +14,14 @@ export const useChainInfo = (options?: useChainInfoOptions) => {
   })
 }
+
+// todo(kon): implement this function
+export const useBlockToDate = ({ height }: { height: number }) => {
+  const { client } = useClient<ExtendedSDKClient>()
+  return { data: new Date() }
+  // return useQuery({
+  //   queryKey: ['blocks', 'blockToDate', height],
+  //   queryFn: () => client.chainService.blockToDate(height),
+  //   ...options,
+  // })
+}
```
Your comment is self-explanatory 🙃

selankon replied:
This is related https://github.com/vocdoni/ui-components/pull/183

elboletaire replied:
On it

Edit: published as 0.0.3

selankon replied:
5161b1f

elboletaire commented on src/components/Transactions/TransactionList.tsx:
```diff
@@ -0,0 +1,36 @@
+
+  const isLoading = isLoadingCount || isLoadingTx
+
+  if (isLoading) {
+    return <LoadingCards />
+  }
+
+  if (!data || data?.transactions.length === 0 || isError) {
+    return <LoadingError error={error} />
+  }
```
Please integrate this into the rendering, as we decided to do yesterday (not conditional rendering when loading or err, since it causes a buggy behavior for subsequent calls).

selankon replied:
526ff36

elboletaire commented on src/utils/strings.ts:
```diff
@@ -4,3 +4,46 @@ export const shortHex = (hex: string) => hex.substring(0, 6) + '...' + hex.subst
+ *
+ * It check if is a string and can be converted to a regex.
+ */
+export const objectB64StringsToHex = (obj: any, ignoreKeys?: string[]): void => {
+  for (const k in obj) {
+    if (ignoreKeys != null && ignoreKeys.indexOf(k) > -1) continue
+    if (typeof obj[k] === 'string' && regex.test(obj[k])) {
+      try {
+        obj[k] = b64ToHex(obj[k])
+      } catch (e) {
+        console.debug("Can't convert (may be not a b64 string)", k, obj[k])
+      }
+    } else if (typeof obj[k] == 'object' && obj[k] !== null) {
+      objectB64StringsToHex(obj[k])
+    }
+  }
+}
```
This should be in `~utils/objects.ts`; move the regex to `constants.ts` to import it from a more generic path (and since it's used by at least two functions here that should be in different util files).

selankon replied:
9898f5d

selankon committed:
Refactor b64ToHex

Move the functions to proper places

selankon committed:
Integrate rendering

Instead of conditional rendering to avoid "buggy behavior for subsequent calls".

It also fixes the paginator url

selankon committed:
Implement search transaction

Implement a search box to show transaction by its number

selankon committed:
Delete todos

Related to pagination

selankon committed:
Bump @vocdoni/chakra-components @vocdoni/extended-sdk

selankon committed:
Implement useBlockToDate

selankon committed:
Fix block to date
